go/state/gostate/go_state_test.go

```diff
@@ -773,6 +792,10 @@ func TestState_All_Live_Operations_May_Cause_Failure(t *testing.T) {
 				if _, err := db.GetHash(); shouldFail && !errors.Is(err, injectedErr) {
 					t.Errorf("operation should fail")
 				}
+				shouldFail = shouldFail || errors.Is(results[7], injectedErr)
```

Comment:
If I read this test setup correctly, this one should use `results[8]` and all subsequent ones should have the index increased by 1. At least in the expectation, the return value set is `results[8]`.

Reply:
you are right, thanks! This test should be probably improved to initialise data in a loop